diff --git a/config/config.go b/config/config.go
index c8c9e1c9..683e74f6 100644
--- a/config/config.go
+++ b/config/config.go
@@ -50,6 +50,7 @@ type General struct {
 	Mode              T.TunnelMode            `json:"mode"`
 	UnifiedDelay      bool                    `json:"unified-delay"`
 	LogLevel          log.LogLevel            `json:"log-level"`
+	Pprof             bool                    `json:"pprof"`
 	IPv6              bool                    `json:"ipv6"`
 	Interface         string                  `json:"interface-name"`
 	RoutingMark       int                     `json:"routing-mark"`
@@ -415,6 +416,7 @@ type RawConfig struct {
 	Mode                          T.TunnelMode            `yaml:"mode" json:"mode"`
 	UnifiedDelay                  bool                    `yaml:"unified-delay" json:"unified-delay"`
 	LogLevel                      log.LogLevel            `yaml:"log-level" json:"log-level"`
+	Pprof                         bool                    `yaml:"pprof" json:"pprof"`
 	IPv6                          bool                    `yaml:"ipv6" json:"ipv6"`
 	ExternalController            string                  `yaml:"external-controller" json:"external-controller"`
 	ExternalControllerRoutingMark int                     `yaml:"external-controller-routing-mark" json:"external-controller-routing-mark"`
@@ -776,5 +778,6 @@ func parseGeneral(cfg *RawConfig) (*General, error) {
 		UnifiedDelay: cfg.UnifiedDelay,
 		Mode:         cfg.Mode,
 		LogLevel:     cfg.LogLevel,
+		Pprof:        cfg.Pprof,
 		IPv6:         cfg.IPv6,
 		Interface:    cfg.Interface,
